pkg/featurechangestopper/featurechangestopper.go

```diff
@@ -0,0 +1,140 @@
+package featurechangestopper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configinformersv1 "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+// TechPreviewChangeStopper calls stop when the value of the featuregate changes from TechPreviewNoUpgrade to anything else
+// or from anything to TechPreviewNoUpgrade.
+type TechPreviewChangeStopper struct {
+	startingTechPreviewState bool
+
+	featureGateLister configlistersv1.FeatureGateLister
+	cacheSynced       []cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+}
+
+// New returns a new cluster version operator.
```

Comment:
nit: should this be "...returns a new TechPreviewChangeStopper." or "...a new TechPreviewNoUpgrade change stopper." or some such?